File: unityProduct/Assets/Script/ViewController/exampleViewController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class exampleViewController : ViewController
{
    public Button btn = null;

    protected override void init()
    {
        base.init();

        btn.onClick.AddListener(onClick);
    }

    protected override void open(object obj)
    {
        base.open(obj);
    }

    public override void close()
    {
        base.close();
    }

    protected override void onTimer()
    {
        base.onTimer();
    }

    private void onClick()
    {
        Debug.Log("onClick");

        ViewController.GetViewController<example2ViewController>().show("‘—‹‹Ž‘—¿");
    }
}

File: FgoProductCocos/assets/script/ViewController/exampleViewController.ts
import { _decorator, Button, Component, Node } from 'cc';
import {ViewController} from './ViewController'
import { example2ViewController } from './example2ViewController';
const { ccclass, property } = _decorator;

@ccclass('exampleViewController')
export class exampleViewController extends ViewController {
    
    @property(Button)
    public btn : Button;

    protected init(): void {
        super.init();
        this.btn.node.on(Button.EventType.CLICK,this.send,this);
    }

    protected open(obj: any): void {
        super.open(obj);
    }

    public close(): void {
        super.close();
    }

    protected onTimer(): void {
        super.onTimer();
    }

    async send()
    {
        var view = await ViewController.GetViewController(example2ViewController);
        view.show();
        view.unitTest();
    }
}


